src/Avalonia.X11/X11Window.cs

```diff
@@ -455,7 +452,7 @@ private bool UpdateScaling()
                     Scaling = newScaling;
                     ScalingChanged?.Invoke(Scaling);
                     SetMinMaxSize(_scaledMinMaxSize.minSize, _scaledMinMaxSize.maxSize);
-                    Resize(oldScaledSize, true);
+                    Resize(oldScaledSize, false);
```

Comment:
@kekekeks passing `true` here was causing an extra `Resized` event with the wrong size, resetting `SizeToContent`. Is the `true` necessary here?

Reply:
It is needed to update the actual window size when DPI is changed.